### PR TITLE
Feat: Add schema map config

### PR DESCRIPTION
**Pull request by z3z1ma**
Not sure what method to use to render a query where this is applied. Once I know this, we can extend the test to validate the mapping is used, and is otherwise a very simple change.

### CODE REVIEW

**Comment by tobymao** [comment]
you need to add this field to the data hash since changing the mapping should trigger a backfill

**Comment by eakmanrq** [comment]
@z3z1ma Can you add this feature to the documentation? My preference is that it is framed as more of an advanced feature and give some examples of where it would be useful. Open to feedback on that though.

**Comment by izeigerman** [comment]
@z3z1ma sorry, I'm not sure I understand your concern here. There is no concept of a "new" model but there is a concept of a "new" snapshot. So whatever you put into a snapshot constitutes a change. 

The way we detect changes is by computing the fingerprint of a **snapshot** which of course includes a model among other things but also covers audits, macros, etc. So I believe putting a physical schema in there should work perfectly fine unless I'm missing something.

File: sqlmesh/core/context.py [hunk 1]
```diff
@@ -525,6 +526,7 @@ def snapshots(self) -> t.Dict[str, Snapshot]:
                 config = self.config_for_model(model)
                 ttl = config.snapshot_ttl
                 project = config.project
+                physical_schema = config.physical_schema_map.get(parse_model_name(model.name)[1])
```
Comment (izeigerman) [inline sqlmesh/core/context.py:529]: Shouldn't we inherit the `physical_schema` from the remote snapshot if one was found in the other branch?

Comment (izeigerman) [reply sqlmesh/core/context.py:529]: So I think you should set `physical_schema` to `config.physical_schema_map.get(parse_model_name(model.name)[1])` regardless of whether the `model.name` is in `remote_snapshots` or not. Which also means you can get rid of `physical_schema = None`.

**Comment by tobymao** [comment]
closing this for now, we may pick this up later, or you're free to @alex-harness

**Pull request closed without merging**